src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -27,14 +31,91 @@
 #include "distributed/metadata_cache.h"
 #include "distributed/metadata_sync.h"
 #include "distributed/multi_join_order.h"
+#include "distributed/pg_dist_node.h"
 #include "distributed/worker_manager.h"
+#include "distributed/worker_transaction.h"
 #include "foreign/foreign.h"
 #include "nodes/pg_list.h"
 #include "utils/builtins.h"
+#include "utils/fmgroids.h"
 #include "utils/lsyscache.h"
 #include "utils/memutils.h"
 
 
+static char * LocalGroupIdUpdateCommand(uint32 groupId);
+static void MarkNodeHasMetadata(char *nodeName, int32 nodePort);
+static char * TruncateTriggerCreateCommand(Oid relationId);
+
+
+PG_FUNCTION_INFO_V1(sync_metadata_to_node);
+
+
+/*
+ * sync_metadata_to_node function creates the metadata in a worker for preparing the
+ * worker for accepting MX-table queries. The function first sets the localGroupId of the
+ * worker so that the worker knows which tuple in pg_dist_node table represents itself.
+ * After that, SQL statetemens for re-creating metadata about mx distributed
+ * tables are sent to the worker. Finally, the hasmetadata column of the target node in
+ * pg_dist_node is marked as true.
+ */
+Datum
+sync_metadata_to_node(PG_FUNCTION_ARGS)
+{
+	text *nodeName = PG_GETARG_TEXT_P(0);
+	int32 nodePort = PG_GETARG_INT32(1);
+	char *nodeNameString = text_to_cstring(nodeName);
+	char *extensionOwner = CitusExtensionOwnerName();
+
+	WorkerNode *workerNode = NULL;
+	char *localGroupIdUpdateCommand = NULL;
+	List *recreateMetadataSnapshotCommandList = NIL;
+	List *dropMetadataCommandList = NIL;
+	List *createMetadataCommandList = NIL;
+
+	EnsureSuperUser();
+
+	PreventTransactionChain(true, "sync_metadata_to_node");
+
+	workerNode = FindWorkerNode(nodeNameString, nodePort);
+
+	if (workerNode == NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
+						errmsg("you cannot sync metadata to a non-existent node"),
+						errhint("First, add the node with master_add_node().")));
```

Comment:
Could give the exact command here, e.g. `... with: SELECT master_add_node(%s,%d)", nodeNameString, nodePort)`